### ISSUE TITLE
Repository 'create' is skipping inherited fields

### ISSUE THREAD

**Issue by AmirTugi** [typeorm/typeorm#78]
I have 2 inherited-style models,
and when I create a new entity using repository.create(JSON),
it just skips the inherited fields.

These are my models

```
@Table()
@TableInheritance("class-table")
export class Document {
    @PrimaryGeneratedColumn()
    id: number;
    @OneToMany(type => DocumentLine, documentLine => documentLine.document)
    documentLines: DocumentLine[];
    @Column()
    dollarRate: number = 0;
    @Column()
    orderBy: string = '';
    @Column()
    comments: string = '';
    @Column()
    subTotal: number = 0;
    @Column()
    vat: number = 0;
    @Column()
    total: number = 0;
    @Column()
    createdBy: string = '';
    @CreateDateColumn()
    createdAt: string;
    @UpdateDateColumn()
    updatedAt: string;
}
```

```
@ClassTableChild()
export class DeliveryNote extends Document {
    @ManyToOne(type => Customer, customer => customer.deliveryNotes)
    customer: Customer;
    @Column()
    invoice: string = '';
}
```

What can cause this?
I have used repository.create so far, and it's a good automatic way for converting the JSON into an entity. I don't want to manually insert the inherited fields now :)

Also, when will the [table inheritance page](https://typeorm.github.io/table-inheritance.html) be ready? It's hard for me to guess how to use them by the [samples](https://github.com/typeorm/typeorm/tree/master/sample)

**Comment by pleerock** [comment]
checkout [this page](https://typeorm.github.io/tables-and-columns.html). But I don't recommend to use class table inheritance because its too complex, have a small value and implementation may have bugs. Alternatively I suggest to simple use abstract tables, inherit them and thats all

**Comment by AmirTugi** [comment]
That's what I did at the beginning, but then I encountered a problem when having a ManyToOne relation and having to point to an abstract table (see below figure)

![2016-11-06 1](https://cloud.githubusercontent.com/assets/8065975/20038177/ad67b256-a437-11e6-8b19-9f8de17d9a94.png)

I get this error:
```
Error: Entity metadata for DocumentLine#document was not found.
    at entityMetadata.relations.forEach.relation (D:\Projects\ArihavAPI\node_modules\typeorm\metadata-builder\EntityMetadataBuilder.js:269:27)
    at PropertyMetadataArgsCollection.forEach (native)
    at entityMetadatas.forEach.entityMetadata (D:\Projects\ArihavAPI\node_modules\typeorm\metadata-builder\EntityMetadataBuilder.js:266:38)
    at Array.forEach (native)
    at EntityMetadataBuilder.build (D:\Projects\ArihavAPI\node_modules\typeorm\metadata-builder\EntityMetadataBuilder.js:265:25)
    at EntityMetadataBuilder.buildFromMetadataArgsStorage (D:\Projects\ArihavAPI\node_modules\typeorm\metadata-builder\EntityMetadataBuilder.js:149:21)
    at Connection.buildMetadatas (D:\Projects\ArihavAPI\node_modules\typeorm\connection\Connection.js:323:18)
    at Connection.<anonymous> (D:\Projects\ArihavAPI\node_modules\typeorm\connection\Connection.js:113:18)
    at next (native)
    at fulfilled (D:\Projects\ArihavAPI\node_modules\typeorm\connection\Connection.js:4:58)
```

Because `Document` is abstract. I do, however, want to keep the `documentLines` relation in `Document`, and so it will be general for any child of it.

These are the models:

```
@AbstractTable()
export class Document {
    @PrimaryGeneratedColumn()
    id: number;
    @OneToMany(type => DocumentLine, documentLine => documentLine.document)
    documentLines: DocumentLine[];
    @Column()
    dollarRate: number = 0;
    @Column()
    orderBy: string = '';
    @Column()
    comments: string = '';
    @Column()
    subTotal: number = 0;
    @Column()
    vat: number = 0;
    @Column()
    total: number = 0;
    @Column()
    createdBy: string = '';
    @CreateDateColumn()
    createdAt: string;
    @UpdateDateColumn()
    updatedAt: string;
}
```

```
@Table()
export class DeliveryNote extends Document {

    @PrimaryGeneratedColumn()
    id: number;

    @ManyToOne(type => Customer, customer => customer.deliveryNotes)
    customer: Customer;

    @Column()
    invoice: string = '';
}
```

```
@Table()
export class DocumentLine {
    @PrimaryGeneratedColumn()
    id: number;
    @ManyToOne(type => Item, item => item.documentLines)
    item: Item;
    @Column()
    itemPartNumber: string = this.item.partNumber;
    @Column()
    itemDescription: string = this.item.description;
    @ManyToOne(type => Document, document => document.documentLines)
    document: Document;
    @Column()
    comments: string = '';
    @Column()
    quantity: number = 0;
    @Column()
    price: number = 0;
    @Column()
    total: number = 0;
    @CreateDateColumn()
    createdAt: string;
    @UpdateDateColumn()
    updatedAt: string;
}
```

**Comment by pleerock** [comment]
fixed and released in `typeorm@0.0.3-alpha.15`, to install it: `npm i typeorm@next`
